core-api/src/main/java/com/optimizely/ab/internal/ConditionUtils.java

```diff
@@ -0,0 +1,171 @@
+/**
+ *
+ *    Copyright 2018, Optimizely and contributors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab.internal;
+
+import com.google.gson.internal.LinkedTreeMap;
+import com.optimizely.ab.config.ProjectConfig;
+import com.optimizely.ab.config.audience.AndCondition;
+import com.optimizely.ab.config.audience.AudienceIdCondition;
+import com.optimizely.ab.config.audience.Condition;
+import com.optimizely.ab.config.audience.NotCondition;
+import com.optimizely.ab.config.audience.OrCondition;
+import com.optimizely.ab.config.audience.UserAttribute;
+import org.json.simple.JSONObject;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ConditionUtils {
+    static public Condition parseConditions(List<Object> rawObjectList) {
```

Comment:
Nice extraction! 🕵

Reply:
Though I still haven't figured out how our JSON parsing works in the Java SDK, so I'll trust @mikeng13 or @aliabbasrizvi to review this part for now